packages/form-js-viewer/test/spec/render/components/form-fields/JSFunctionField.spec.js

```diff
@@ -0,0 +1,163 @@
+import {
+  render,
+  waitFor
+} from '@testing-library/preact/pure';
+
+import { JSFunctionField } from '../../../../../src/render/components/form-fields/JSFunctionField';
+
+import { MockFormContext } from '../helper';
+
+import { act } from 'preact/test-utils';
+
+import {
+  createFormContainer
+} from '../../../../TestHelper';
+
+let container;
+
+describe('JSFunctionField', function() {
+
+  beforeEach(function() {
+    container = createFormContainer();
+  });
+
+
+  afterEach(function() {
+    container.remove();
+  });
+
+
+  it('should evaluate with setValue', async function() {
+
+    // given
+    const onChangeSpy = sinon.spy();
+    const field = defaultField;
+    const passedData = { value : 42 };
+
+    const services = {
+      expressionLanguage: {
+        isExpression: () => true,
+        evaluate: () => {
+          return passedData;
+        }
+      }
+    };
+
+    // when
+    await act(() => {
+      createJSFunctionField({ field, onChange: onChangeSpy, services });
+    });
+
+    // then
+    await waitFor(() => {
+      expect(onChangeSpy).to.be.calledOnce;
+      expect(onChangeSpy).to.be.calledWith({ field, value: 42 });
+    });
+
+  });
+
+
+  it('should evaluate with return', async function() {
+
+    // given
+    const onChangeSpy = sinon.spy();
+    const field = {
+      ...defaultField,
+      jsFunction: 'return data.value'
+    };
+    const passedData = { value : 42 };
+
+    const services = {
+      expressionLanguage: {
+        isExpression: () => true,
+        evaluate: () => {
+          return passedData;
+        }
+      }
+    };
+
+    // when
+    act(() => {
+      createJSFunctionField({ field, onChange: onChangeSpy, services });
+    });
+
+    // wait for the iframe to compute the expression and pass it back
+    await new Promise(r => setTimeout(r, 100)).then(() => {
```

Comment:
Please also replace the other occurrences with the `waitFor` I mentioned in the other comments